File: Algorithms/Sorting/gnome_sort.c
#include <stdio.h>

void
gnome_sort (int* seq, int n)
{
  int i = 0;
  while (i < n)
    {
      if (i == 0 || seq[i] >= seq[i - 1])
        i++;
      else
        {
          int tmp = seq[i];
          seq[i] = seq[i - 1];
          seq[i - 1] = tmp;
          i--;
        }
    }
}


int
main ()
{
  int seq[] = {3, 5, 2, 4, 1};
  int n = 5;

  gnome_sort (seq, n);

  for (int i = 0; i < n; ++i)
    printf ("%d ", seq[i]);
  printf ("\n");
}

File: Algorithms/Sorting/gnome_sort.rs
fn gnome_sort(seq: &mut [u8; 5]) {
    let mut i: usize = 0;
    while i < seq.len() {
        if i == 0 || seq[i] >= seq[i - 1] {
            i += 1;
        } else {
            seq.swap(i, i - 1);
            i -= 1;
        }
    }
}

fn main() {
    let mut seq: [u8; 5] = [3, 5, 2, 4, 1];

    gnome_sort(&mut seq);
    
    for i in &seq {
        print!("{} ", i);
    }
    print!("\n");
}

    
